### PR TITLE
Fixup arm and arm64 builds; Refactor Makefile;

### PR DESCRIPTION
**Pull request by brezerk**
HW: Rock64 (arm64v8)
OS: Gentoo

Hi

Thank you for nice DevOps tool. I was looking forward to build if for my arm64 k8s cluster and got some issues.

Here you are the changes I had to make to compile the binaries and build the images.

Also, i have noticed comments/discussions around #1761 , #1411 and #1474 so i have tried to update Makefile accordingly.
 
Retry for #1942 :)

### CODE REVIEW

**Comment by brezerk** [comment]
Okay. I finally got tests passed :) jfyi: @2opremio

File: bin/upload-binaries [hunk 1]
```diff
@@ -8,15 +8,27 @@ GITHUB_USER=${GITHUB_USER:-"${CIRCLE_PROJECT_USERNAME}"}
...
+	github-release upload \
+		--user ${GITHUB_USER} \
+		--repo ${GITHUB_REPO} \
+		--tag ${GITHUB_TAG} \
+		--name "fluxctl_${os}_${arch}" \
+		--file "build/fluxctl_${os}_${arch}"
+	echo "* Finished pushing fluxctl_${os}_${arch} for ${GITHUB_TAG}"
+}
+
 for arch in amd64; do
```
Comment (2opremio) [inline bin/upload-binaries]: There is no point in having a loop for just an item, please inline it.

File: bin/upload-binaries [hunk 2]
```diff
@@ -8,15 +8,27 @@ GITHUB_USER=${GITHUB_USER:-"${CIRCLE_PROJECT_USERNAME}"}
...
-			--repo ${GITHUB_REPO} \
-			--tag ${GITHUB_TAG} \
-			--name "fluxctl_${os}_${arch}" \
-			--file "build/fluxctl_${os}_${arch}"
-		echo "* Finished pushing fluxctl_${os}_${arch} for ${GITHUB_TAG}"
+		do_publish $os $arch
+	done
+done
+for arch in arm arm64; do
+	for os in linux; do
```
Comment (2opremio) [inline bin/upload-binaries]: There is no point in having a loop for just an item, please inline it.

**Comment by 2opremio** [comment]
Great work! it's shaping up nicely!

Wanna bite the bullet and create multi-architecture images when pushing from CircleCI? It can be done in a separate PR if you want.

The advantage of that is that people interested in ARM images won't have to build locally.

File: bin/upload-binaries (hunk 1 above)
Comment (brezerk) [reply bin/upload-binaries]: done

File: bin/upload-binaries (hunk 2 above)
Comment (brezerk) [reply bin/upload-binaries]: done

**Comment by brezerk** [comment]
Yup, I'd love to see official multi-architecture images too.
However i'll need couple days to get acquainted with circleci tool and `binfmt_misc` approach used in prometheus/prometheus#5031

I'd vote for a separate PR :)

File: site/building.md [hunk 3]
```diff
@@ -21,6 +21,13 @@ $ make
 
 This makes Docker images, and installs binaries to $GOPATH/bin.
 
+Note: Default target architecture is amd64. If you would like to try to build Docker images
```
Comment (2opremio) [inline site/building.md]:
```suggestion
Note: The default target architecture is amd64. If you would like to try to build Docker images
```

**Comment by 2opremio** [comment]
> Yup, I'd love to see official multi-architecture images too.

Awesome!! Looking forward to that work.

Almost ready to merge, just a small nitpick.

File: site/building.md (hunk 3 above)
Comment (brezerk) [reply site/building.md]: Done. Thank you!

**Comment by squaremo** [comment]
This is thorough and neatly done -- thanks for giving it another go @brezerk

**Pull request merged**